src/server/services/collection.service.ts

```diff
@@ -183,11 +208,43 @@ export const saveItemInCollections = async ({
   user: SessionUser;
   input: AddCollectionItemInput;
 }) => {
+  const inputToCollectionType: Record<string, CollectionType> = {
+    modelId: CollectionType.Model,
+    articleId: CollectionType.Article,
+    imageId: CollectionType.Image,
+    postId: CollectionType.Post,
+  };
+
+  const itemKey = Object.keys(inputToCollectionType).find((key) => input.hasOwnProperty(key));
+
+  if (itemKey && inputToCollectionType.hasOwnProperty(itemKey)) {
+    const type: CollectionType = inputToCollectionType[itemKey];
+    // check if all collections match the Model type
+    const collections = await dbRead.collection.findMany({
+      where: {
+        id: { in: collectionIds },
+        OR: [
+          {
+            type: null,
+          },
+          {
+            type,
+          },
+        ],
+      },
+    });
+
+    if (collections.length !== collectionIds.length) {
+      throw new Error('Collection type mismatch');
```

Comment:
Let's throw an error with the proper http code
```suggestion
      throw throwBadRequestError('Collection type mismatch');
```